src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java

```diff
@@ -71,4 +82,22 @@ private Long checkClicksAmountOnShortUrl(UUID urlId) {
             return incMaxClickValue;
         }
     }
+
+    public List<UrlDto> findMostPopularUrls() {
+        PageRequest pageRequest = PageRequest.of(0, 3);
+        List<MostPopularUrlResult> results = urlRepository.findMostPopularUrls(pageRequest);
+
+        List<UrlDto> popularUrls = new ArrayList<>();
+        for (MostPopularUrlResult result : results) {
```

Comment:
Looks like a lot of logic.
Think how to simplify it, or at least move it to a separate method/class.